### ISSUE TITLE
Regression after commit b4018c22f8fe8371bd6845d0cd62cebea54ce012

### ISSUE THREAD

**Issue by Hekzory** [ansible/ansible-lint#4234]
##### Summary

Before I could run ansible-lint github action in non-root directory via working_directory variable (actual config in screenshot below). It would also read my ansible.cfg, install requirements, etc.

Now I can't. If this is not a regression but a breaking change, how can I achieve it now?

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

GitHub Actions, ansible-lint@main

##### STEPS TO REPRODUCE

Here is my config

![image](https://github.com/ansible/ansible-lint/assets/36638457/a9daf4e3-2dd9-49c7-9c0a-b63138541df3)

##### Desired Behavior

It works

##### Actual Behavior

Here is a screenshot of the error:

![image](https://github.com/ansible/ansible-lint/assets/36638457/535640fd-8bc1-4a9c-847c-24398ee877a7)

**Comment by ostefek99** [comment]
I do observe the same issue. The GitHub Action is working in version `v24.6.0` but it breaks in `v24.6.1`.

Working version `v24.6.0`:
![image](https://github.com/ansible/ansible-lint/assets/60153717/bd4126e8-bf81-4297-8158-89478773d6d8)

Stopped working in `v24.6.1`:
![image](https://github.com/ansible/ansible-lint/assets/60153717/06c8ee5b-c442-442a-9d2b-7332fc8e0c6f)

**Comment by solomonjames** [comment]
There is an assumption the working directory is one with a `.git` folder. 

Current work around is to create a fake `.git` directory in your working directory.

```yaml
      - name: Add fake .git dir
        run: mkdir -p <working_directory>/.git
```

**Comment by drew-viles** [comment]
See, my problem is similar but I'm using this in a composite action and so due to the way it's been implemented, the value of `${{ github.action_ref }}` is my actions ref: `https://raw.githubusercontent.com/ansible/ansible-lint/add_linters/.config/requirements-lock.txt` - Note the add_linters bit.

I've raised a test fix in a new PR to enable an override for this - https://github.com/ansible/ansible-lint/pull/4481.

I have also not had to add the recommendation above to create a .git dir even though my working_dir is `./ansible` so this PR may also fix that. Cannot garuantee it though as I've not had that particular problem to be able to test the fix of it.
